lib/Config.re

```diff
@@ -0,0 +1,66 @@
+[@deriving show]
+type t = {
+  sandboxPath: Fpath.t,
+  storePath: Fpath.t,
+  localStorePath: Fpath.t
+};
+
+let storeInstallTree = "i";
+
+let storeBuildTree = "b";
+
+let storeStageTree = "s";
+
+let storeVersion = "3";
+
+let maxStorePaddingLength = {
+  /* This is restricted by POSIX, Linux enforces this but macOS is more
+   * forgiving. */
+  let maxShebangLength = 127;
+  /* We reserve that amount of chars from padding so ocamlrun can be placed i
+   * shebang lines */
+  let ocamlrunStorePath = "ocaml-n.00.000-########/bin/ocamlrun";
+  maxShebangLength
+  - String.length("!#")
+  - String.length(
+      "/"
+      ++ storeVersion
+      ++ "/"
+      ++ storeInstallTree
+      ++ "/"
+      ++ ocamlrunStorePath
+    );
+};
+
+/**
```

Comment:
Down below:
This is the main suboptimal printing that only occurs with `ppx` extension point sugar.
Normally, local opens of this form would be printed as:

```reason
{
  open Run;
  let%bind prefixPath =
    switch prefixPath {
    | Some(v) => Ok(v)
    | None =>
      let%bind home = Bos.OS.Dir.user();
      Ok(home / ".esy");
    };
}
```

We just need to special case that same rule for ppx extension point sugar.

Reply:
Heh, the same bug was with my fork — if `let ... = do ...` was the first expr in a sequence then the local open was printed like that.